crates/evm/evm/src/inspectors/stack.rs

```diff
@@ -726,6 +726,18 @@ impl<'a, DB: DatabaseExt> Inspector<DB> for InspectorStackRefMut<'a> {
 
         ecx.journaled_state.depth += self.in_inner_context as usize;
         if let Some(cheatcodes) = self.cheatcodes.as_deref_mut() {
+            // Handle mocked functions, replace bytecode address with mock if matched.
+            if let Some(mocks) = cheatcodes.mocked_functions.get(&call.target_address) {
+                if let Some(target) = mocks.get(&call.input) {
+                    call.bytecode_address = *target;
+                } else {
+                    // Check if we have a catch-all mock set for selector.
+                    if let Some(target) = mocks.get(&call.input.slice(..4)) {
```

Comment:
can collapse into one `get().or_else(|| )`, also this panics if `call.input.len() < 4`, use `call.input.get(..4)`

Reply:
changed in https://github.com/foundry-rs/foundry/pull/8807/commits/7e69723609d13ceb3ce744c0f7bbdd3e156f1296